### DISCUSSION TITLE
Problem using TemplatedControl

### DISCUSSION THREAD

**Discussion by Kruno313** [AvaloniaUI/Avalonia#16447] [Q&A] [answered]
I have created simple TemplatedControl and use it in the view, but nothing is rendered, except empty space if I set Height=20. I'm using latest Avalonia 11.1.0.
Older tutorials seems to have different .axaml code in the generated file, starting with '<Styles ..'.
What I have got using _new Avalonia TemplatedControl_ is this: 
```html
<ResourceDictionary xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:controls="using:eCapConfigurator.Controls">
  <!--
    Additional resources 
    Using Control Themes:         https://docs.avaloniaui.net/docs/basics/user-interface/styling/control-themes
    Using Theme Variants:         https://docs.avaloniaui.net/docs/guides/styles-and-resources/how-to-use-theme-variants
  -->
    <Design.PreviewWith>
    <StackPanel Width="400" Spacing="10">      
        <StackPanel Background="{DynamicResource SystemRegionBrush}">
          <controls:TemplatedControl1 />
        </StackPanel>
    </StackPanel>
  </Design.PreviewWith>
 
  <ControlTheme x:Key="{x:Type controls:TemplatedControl1}" TargetType="controls:TemplatedControl1">
    <Setter Property="Template">
      <ControlTemplate>
        <TextBlock Text="Templated Control" />
      </ControlTemplate>
    </Setter>
  </ControlTheme>
</ResourceDictionary>
``` 
Searching for the error in web brings the need for `<StyleInclude Source = "/Controls/TemplatedControl1.axaml" />` in App.axaml  (see [Templated Control doesn't appear #6475](https://github.com/AvaloniaUI/Avalonia/issues/6475) ), but that produces error for my templatad control: 
> Resource "avares://project/Controls/TemplatedControl1.axaml" is defined as "Avalonia.Controls.ResourceDictionary" type in the "project" assembly, but expected "Avalonia.Styling.IStyle".
The Avalonia itself lacks documentation how to use TemplatedControl.

Is the template for TemplatedControl wrong or do I have to use it some other way?

**Comment by workgroupengineering** [comment, marked as answer]
Should use like this

```xaml
  <Application.Resources>
    <ResourceDictionary>
      <ResourceDictionary.MergedDictionaries>
        <ResourceInclude Source="/Controls/TemplatedControl1.axaml"/>
      </ResourceDictionary.MergedDictionaries>
      <x:Double x:Key="SmallIconSize">16</x:Double>
      <x:Double x:Key="NormalIconSize">24</x:Double>
      <x:Double x:Key="LargeIconSize">24</x:Double>
    </ResourceDictionary>
  </Application.Resources>
```

**Reply by Kruno313** [reply]
Thank You!
It works with this in the App.axaml.
